src/coreclr/src/jit/morph.cpp

```diff
@@ -18213,6 +18319,57 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
         }
     }
 
+    //------------------------------------------------------------------------
+    // MorphLocalAddress: Change a tree that represents a local variable address
+    //    to a single LCL_VAR_ADDR or LCL_FLD_ADDR node.
+    //
+    // Arguments:
+    //    val - a value that represents the local address
+    //
+    void MorphLocalAddress(const Value& val)
+    {
+        assert(val.IsAddress());
+        assert(val.Node()->TypeIs(TYP_BYREF, TYP_I_IMPL));
+        assert(m_compiler->lvaVarAddrExposed(val.LclNum()));
+
+        LclVarDsc* varDsc = m_compiler->lvaGetDesc(val.LclNum());
+
+        if (varDsc->lvPromoted || varDsc->lvIsStructField || m_compiler->lvaIsImplicitByRefLocal(val.LclNum()))
+        {
+            // Ignore promoted and "implict by ref" variables for now,
+            // they require additional changes in subsequent phases.
```

Comment:
Should this also have a "TODO"? Or perhaps instead we need a general issue to capture the various improvements that are outstanding in this area?

Reply:
Yes, I think a TODO is reasonable, both here and on the other comment above.

I already added a list with the various `ADDR` use cases to the existing `ADDR` issue so it would make sense to also add a list of TODOs as well.